libmamba/src/api/list.cpp

```diff
@@ -34,6 +35,31 @@ namespace mamba
             return a.name < b.name;
         }
 
+        // This is more or less an implementation of `util::rstrip` specific to this use case
+        // (for printing purposes), but using `std::string` instead of `std::string_view`
+        // `util::rstrip` is not used here because it leads to an UB,
+        // `using non owned/tracked strings from Channel (& co) and PackageInfo
```

Comment:
Not sure to understand why `util::rstrip` leads to UB. Is it still the case if you assign its result to a `std::string`?
Anyway, if we keep this implementation, we should open an issue to try to fix the UB and use a single implementation of rstrip.

Reply:
I meant using the funciton degined in `util::string` and assign the final result of the recursive call to a string. The function returns temporary string_view passed by copies, so there hsould not be any dangling temporary issues.

Reply:
Well doing:
```
std::string res = std::string(util::rstrip(util::rstrip(util::rstrip(util::rstrip(full_str, filename), "/"), platform), "/"));
```
instead of using the reimplemented `rstrip` in `strip_from_filename_and_platform` definitely leads to UB, if that's what you meant.

Reply:
yep that's what I meant :( OK let's merge as is and open an issue to fix that UB later